swerve_animator_node:
  ros__parameters:
    skip_factor: 20
    speedup_factor: 1.5

swerve_mpc_planner_node:
  ros__parameters:
    publish_intermediate_solutions: true
    ipopt_verbosity: 5
    max_iterations: 750

    time_horizon: 5.0
    dt: 0.05
    num_swerve_modules: 4

    chassis_width: 16.5
    wheel_base_width: 12.5
    wheel_diam: 2.75
    wheel_width: 1.0
    robot_mass: 10.0
    robot_inertia: 0.14868
    steering_inertia: 0.00005
    wheel_constraint_tolerance: 0.001

    steering_ratio: 0.295454545
    wheel_ratio: 0.633116883
    motor_model_free_speed_rpm: 700.0
    motor_model_stall_torque_nm: 0.6
    motor_torque_limit_nm: 0.35
    motor_speed_limit_rpm: 650.0

    default_state_weights:
      - 100000.0 # base_pose_x
      - 100000.0 # base_pose_y
      - 100000.0 # base_pose_theta
      - 0.0 # base_vel_x
      - 0.0 # base_vel_y
      - 0.0 # base_vel_theta
      - 0.0 # base_accel_x
      - 0.0 # base_accel_y
      - 0.0 # base_accel_theta

    default_joint_weights:
      - 0.0 # steering_angle
      - 0.5 # steering_vel
      - 0.00001 # steering_accel
      - 0.01 # wheel_vel
      - 0.5 # wheel_torque
      - 0.5 # lateral_force
      - 0.00001 # m1_velocity
      - 0.00001 # m2_velocity
      - 0.00001 # m1_torque
      - 0.00001 # m2_torque
      - 0.00001 # m1_voltage
      - 0.00001 # m2_voltage
